### DISCUSSION TITLE
Upgrade template & utility functions

### DISCUSSION THREAD

**Discussion by taqtiqa-mark** [bashly-framework/bashly#272] [General]
Low hanging fruit:

[one resource](https://github.com/ralish/bash-script-template/blob/main/template.sh)
many others....

**Comment by DannyBen** [comment]
I dont know what to do with this.

**Comment by taqtiqa-mark** [comment]
There is a wealth of experience and subtle corner cases built into these utility functions, e.g. trap handling and running as root etc.

A reference/resource as Bashly heads toward 1.0?

**Comment by DannyBen** [comment]
> A reference/resource as Bashly heads toward 1.0?

In this case, thanks.
Lets keep it as an issue for now, since it might help the other open issues. 
Afterwards I will probably convert it to a discussion.

Also - this might contain some functions that can be added as internal libraries for `bashly add ...`
